src/clj/game/cards/upgrades.clj

```diff
@@ -23,6 +23,18 @@
               :yes-ability {:msg (msg "lower the rez cost of " (:title target) " by 3 [Credits]")
                             :effect (effect (rez eid target {:cost-bonus -3}))}}}]})
 
+(defcard "Anoetic Void"
+  {:events [{:event :successful-run
+             :optional
+             {:prompt "Pay 2 [Credits] and trash 2 cards from HQ to end the run?"
+              :req (req (and (can-pay? state side eid card nil [:credit 2 :trash-from-hand 2])
+                             this-server))
+              :yes-ability
+              {:async true
+               :msg "pay 2 [Credits] and trash 2 cards from HQ to end the run"
+               :effect (req (wait-for (pay state :corp card [:credit 2 :trash-from-hand 2])
```

Comment:
This should be `:cost [:credit 2 :trash-from-hand 2]` and then the cost part can be removed from `:msg`, allowing for spending hosted credits to show up in the log.